src/main/java/net/minecraftforge/items/CapabilityItemHandler.java

```diff
@@ -0,0 +1,283 @@
+import net.minecraftforge.fml.common.eventhandler.EventPriority;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+import net.minecraftforge.items.wrapper.InvWrapper;
+import net.minecraftforge.items.wrapper.SidedInvWrapper;
+
+import java.util.concurrent.Callable;
+
+public class CapabilityItemHandler {
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_CHEST = new ResourceLocation("Forge", "ItemHandler_VanillaWrapperChest");
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_SIDED = new ResourceLocation("Forge", "ItemHandler_VanillaWrapperISided");
+    private static final ResourceLocation RESOURCE_LOCATION_VANILLA_SIMPLE = new ResourceLocation("Forge", "ItemHandler_VanillaWrapper");
+    private static final ResourceLocation RESOURCE_LOCATION_PASSTHROUGH = new ResourceLocation("Forge", "ItemHandler_PassThrough");
+    @CapabilityInject(IItemHandler.class)
+    public static Capability<IItemHandler> ITEM_HANDLER_CAPABILITY = null;
+    public static CapabilityItemHandler INSTANCE = new CapabilityItemHandler();
+
+    static
+    {
+        CapabilityManager.INSTANCE.register(IItemHandler.class, new Capability.IStorage<IItemHandler>() {
+            @Override
+            public NBTBase writeNBT(Capability<IItemHandler> capability, IItemHandler instance, EnumFacing side)
+            {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            public void readNBT(Capability<IItemHandler> capability, IItemHandler instance, EnumFacing side, NBTBase nbt)
+            {
+                throw new UnsupportedOperationException();
+            }
+        }, new Callable<IItemHandler>() {
+            @Override
+            public IItemHandler call() throws Exception
+            {
+                // I have no idea how to implement a default item handler
+                // maybe the empty item handler?
+                throw new UnsupportedOperationException();
+            }
+        });
+    }
+
+    private CapabilityItemHandler()
+    {
+
+    }
+
+    @SuppressWarnings("unused")
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public void attachVanillaTileEntityWrappers(AttachCapabilitiesEvent.TileEntity event)
+    {
+        TileEntity te = event.getTileEntity();
+        // If the TE implements IItemHandler, automatically add it to the capability map.
+        // Suggestion: Extend this automatically to all capability interfaces?
+        if (te instanceof IItemHandler)
+        {
+            final IItemHandler handler = (IItemHandler) te;
+            event.addCapability(RESOURCE_LOCATION_PASSTHROUGH, new PassThruWrapper(handler));
+        }
+
+        if (te.getClass() == TileEntityChest.class)
+        {
+            final TileEntityChest chest = (TileEntityChest) te;
+            event.addCapability(RESOURCE_LOCATION_VANILLA_CHEST, new VanillaWrapperChest(chest));
+        }
```

Comment:
We do not want two handlers for the same capability on a TE. Also I would prefer to see this implemented in a patch file instead of a attachment. So we garentee we have control over it.